driver-core/src/main/com/mongodb/client/model/expressions/Expression.java

```diff
@@ -249,19 +254,20 @@ public interface Expression {
      * @return the resulting value.
      * @param <T> the type of the values of the resulting map.
      */
-    <T extends Expression> MapExpression<T> isMapOr(MapExpression<? extends T> other);
+    <T extends Expression> MapExpression<T> isMapOr(MapExpression<@MqlUnchecked(TYPE_ARGUMENT) ? extends T> other);
```

Comment:
Here and in `DocumentExpression.getMap(String, MapExpression)` we should annotate the type argument in the type that we return, not in the type that we accept (like it's in `isArrayOr`).

Reply:
Moved to next PR due to sensitive rebase.